controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -239,14 +241,21 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 	} else if r.StorageClassRequest.Spec.Type == "sharedfilesystem" {
 		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
 		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephFilesystemSubVolumeGroup" {
-				r.cephFilesystemSubVolumeGroup.Name = res.Name
-				break
-			}
+
+		cephFilesystemSubVolumeGroupList := &rookCephv1.CephFilesystemSubVolumeGroupList{}
+		listOptions := &client.MatchingLabels{
+			controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+			controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
 		}
-		if r.cephFilesystemSubVolumeGroup.Name == "" {
-			r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s-%s", r.storageConsumer.Name, generateUUID())
+		cephFilesystemSubVolumeGroup, err := fetchCephFilesystemSubVolumeGroupByOwnerReference(r.ctx, r.Client, r.StorageClassRequest.ObjectMeta, cephFilesystemSubVolumeGroupList, client.InNamespace(r.OperatorNamespace), listOptions)
```

Comment:
1. Don't return  
2. Can we have this be more generic and utilize the controller struct?

 ```suggestion
		err := listByOwner(&cephFilesystemSubVolumeGroupList, r.storageClassRequest,  client.InNamespace(r.OperatorNamespace))
```